app-drawer-layout/app-drawer-layout.html

```diff
@@ -190,6 +201,27 @@
         return forceNarrow ? '' : '(max-width: ' + responsiveWidth + ')';
       },
 
+      _updateHeaderLayoutDisabledState: function() {
+        var drawer = this.drawer;
+        if (!this.drawer) {
+          return;
+        }
+
+        // Set the disabled state on all <app-header-layout> elements distributed in the content
+        // insertion point. Does not change the disabled state for any <app-header-layout> elements
+        // inside the drawer.
+        var disableScroll = !drawer.persistent && drawer.opened;
+        var contentDistributedNodes = Polymer.dom(this.$.content).getDistributedNodes();
+        for (var i = 0; i < contentDistributedNodes.length; i++) {
+          var node = contentDistributedNodes[i];
+          if (node.tagName === 'APP-HEADER-LAYOUT') {
+            node.disabled = disableScroll;
+          }
+          Array.prototype.forEach.call(node.querySelectorAll('app-header-layout'), function(node) {
+            node.disabled = disableScroll;
+          });
+        };
```

Comment:
I'm in for this approach. Would it be any timing issue?  e.g. the attached callback of the app-header-layout fires before this app-drawer-layout is upgraded?